include/swift/AST/DiagnosticsSema.def

```diff
@@ -613,6 +613,12 @@ ERROR(invalid_redecl,none,"invalid redeclaration of %0", (DeclName))
 NOTE(invalid_redecl_prev,none,
      "%0 previously declared here", (DeclName))
 
+WARNING(deprecated_redecl_by_optionality, none,
```

Comment:
Should this be an error in Swift 5 mode?

Reply:
Once IUOs are removed from the type system we'll never reach the code emitting the warning but will fall through to the error about redeclaration below.